sprocs/check_assessment_access.sql

```diff
@@ -71,9 +77,35 @@ BEGIN
         aar.assessment_id = check_assessment_access.assessment_id
         AND caar.authorized
     ORDER BY
+        aar.submittable DESC,
```

Comment:
When the submittable rule is used, the available credit should be 0 (because if one cannot submit answers, it doesn't make sense to have credit be available). However, it is possible for someone to mistakenly set the available credit to a nonzero value. For example:

```
"allowAccess": [
    {
        "startDate": "2021-01-01T00:00:01",
        "endDate": "2021-01-01T23:59:59",
        "credit": 75
    },
    {
        "submittable": false,
        "credit": 100
    }
]
```

If I did not give submittable assessments priority, then the access rule with `submittable` being `false` would always be the active access rule. The assessment would never be submittable.

I considered just putting a warning in the documentation saying that if `submittable` is set to `false`, then the available `credit` must be 0. However, I felt that I should do more to handle the potential mistake. Ideally, I'd throw some sort of error if there is an access rule with `submittable` being `false` and `credit` being nonzero, but I don't see an easy way to do that. Should I just put a warning in the documentation and leave it at that?

Reply:
That is an interesting point, but I think the proper course of action is not to give priority to submittable, it's to ensure that non-submittable rules don't have credit. This could be done with something on sync that rejects a submittable rule with non-zero credit (or that gives a warning and sets the credit to zero).